engine/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SqlParser.scala

```diff
@@ -239,17 +244,20 @@ object SqlParser extends AbstractSparkSQLParser with DataTypeParser {
   protected lazy val notExpression: Parser[Expression] =
     NOT.? ~ comparisonExpression ^^ { case maybeNot ~ e => maybeNot.map(_ => Not(e)).getOrElse(e) }
 
+  protected lazy val pointWrapper: Parser[Expression] =
+    POINT ~> "(" ~> repsep(termExpression, ",")  <~ ")" ^^ {
+      case points => PointWrapperExpression(points)
+    }
+
+  protected lazy val pointExpression: Parser[Expression] =
```

Comment:
You should not define an expression like this. This is because term expression is not necessarily a point, but can be a point. You should make this choice at exactly at where you need to use it, but not define something like this (namely `pointExpression`) .